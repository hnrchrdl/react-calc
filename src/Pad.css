.pad {
  margin: 10px;
  height: 50px;
  width: 50px;
  border: 1px solid #eee;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}
.pad:hover {
  border: 1px solid salmon;
}
