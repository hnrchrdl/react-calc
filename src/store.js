import { createStore, combineReducers } from 'redux';
import calcReducer from './calculator';

export default createStore(combineReducers({ calculator: calcReducer }));
