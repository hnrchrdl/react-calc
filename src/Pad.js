import React from 'react';
import './Pad.css';
import { actionCreators as calculatorActionCreators } from './calculator';
import { connect } from 'react-redux';

export class UnwrappedPad extends React.Component {
  handleClick = () => {
    const { value, undo, reset, ok, addCalcStep } = this.props;
    switch (value) {
      case 'undo':
        undo();
        break;
      case 'reset':
        reset();
        break;
      case 'ok':
        ok();
        break;
      default:
        return addCalcStep(value);
    }
  };
  render() {
    return (
      <div className="pad" onClick={this.handleClick}>
        {this.props.display}
      </div>
    );
  }
}

const mapStateToProps = null;
const mapDispatchToProps = dispatch => ({
  addCalcStep: val => dispatch(calculatorActionCreators.addExpression(val)),
  undo: () => dispatch(calculatorActionCreators.undo()),
  reset: () => dispatch(calculatorActionCreators.reset()),
  ok: () => dispatch(calculatorActionCreators.ok())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UnwrappedPad);
