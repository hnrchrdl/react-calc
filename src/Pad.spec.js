import React from 'react';
import Pad, { UnwrappedPad } from './Pad';
import store from './store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it('matches snapshot without props', () => {
  const component = renderer.create(<Pad store={store} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches snapshot with props', () => {
  const component = renderer.create(
    <Pad store={store} display={'test'} value={1} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('adds calculation steps', () => {
  const mockAddCalcStep = jest.fn();
  const mockUndo = jest.fn();
  const mockReset = jest.fn();
  const mockOk = jest.fn();
  const value = 1;
  const element = shallow(
    <UnwrappedPad
      store={store}
      value={value}
      addCalcStep={mockAddCalcStep}
      undo={mockUndo}
      reset={mockReset}
      ok={mockOk}
    />
  );
  element.simulate('click');
  expect(mockUndo).not.toHaveBeenCalled();
  expect(mockReset).not.toHaveBeenCalled();
  expect(mockOk).not.toHaveBeenCalled();
  expect(mockAddCalcStep).toHaveBeenCalledWith(value);
});

it('calls reset', () => {
  const mockAddCalcStep = jest.fn();
  const mockUndo = jest.fn();
  const mockReset = jest.fn();
  const mockOk = jest.fn();
  const element = shallow(
    <UnwrappedPad
      store={store}
      display={'reset'}
      value={'reset'}
      addCalcStep={mockAddCalcStep}
      undo={mockUndo}
      reset={mockReset}
      ok={mockOk}
    />
  );
  element.simulate('click');
  expect(mockUndo).not.toHaveBeenCalled();
  expect(mockReset).toHaveBeenCalled();
  expect(mockOk).not.toHaveBeenCalled();
  expect(mockAddCalcStep).not.toHaveBeenCalled();
});

it('calls undo', () => {
  const mockAddCalcStep = jest.fn();
  const mockUndo = jest.fn();
  const mockReset = jest.fn();
  const mockOk = jest.fn();
  const element = shallow(
    <UnwrappedPad
      store={store}
      display={'undo'}
      value={'undo'}
      addCalcStep={mockAddCalcStep}
      undo={mockUndo}
      reset={mockReset}
      ok={mockOk}
    />
  );
  element.simulate('click');
  expect(mockUndo).toHaveBeenCalled();
  expect(mockReset).not.toHaveBeenCalled();
  expect(mockOk).not.toHaveBeenCalled();
  expect(mockAddCalcStep).not.toHaveBeenCalled();
});

it('calls ok', () => {
  const mockAddCalcStep = jest.fn();
  const mockUndo = jest.fn();
  const mockReset = jest.fn();
  const mockOk = jest.fn();
  const element = shallow(
    <UnwrappedPad
      store={store}
      display={'ok'}
      value={'ok'}
      addCalcStep={mockAddCalcStep}
      undo={mockUndo}
      reset={mockReset}
      ok={mockOk}
    />
  );
  element.simulate('click');
  expect(mockUndo).not.toHaveBeenCalled();
  expect(mockReset).not.toHaveBeenCalled();
  expect(mockOk).toHaveBeenCalled();
  expect(mockAddCalcStep).not.toHaveBeenCalled();
});
