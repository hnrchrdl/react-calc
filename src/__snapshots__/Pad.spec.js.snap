// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with props 1`] = `
<div
  className="pad"
  onClick={[Function]}
>
  test
</div>
`;

exports[`matches snapshot without props 1`] = `
<div
  className="pad"
  onClick={[Function]}
/>
`;
